source/index.js

```diff
@@ -56,6 +57,32 @@ module.exports = async (input = 'patch', options) => {
 	const hasLockFile = fs.existsSync(path.resolve(rootDir, 'package-lock.json')) || fs.existsSync(path.resolve(rootDir, 'npm-shrinkwrap.json'));
 	const isOnGitHub = options.repoUrl && hostedGitInfo.fromUrl(options.repoUrl).type === 'github';
 
+	let published = false;
+
+	const rollback = onetime(async () => {
+		console.log('\nPublish failed. Rolling back to the previous state...');
+
+		const lastCommit = await execa.stdout('git', ['log', '-1', '--pretty=%B']);
```

Comment:
I thought about using the `getNewVersion` function (in `source/version.js`) instead of the Git command, but wasn't sure if it would help in this case.

Reply:
I think using Git here is the right call.